common/persistence/sql/sqlMetadataPersistenceV2.go

```diff
@@ -201,99 +218,84 @@ func lockMetadata(tx *sqlx.Tx) error {
 	return nil
 }
 
-func (m *sqlMetadataManager) CreateDomain(request *persistence.CreateDomainRequest) (*persistence.CreateDomainResponse, error) {
-	// Disallow creating more than one domain with the same name, even if the UUID is different.
-	resp, err := m.GetDomain(&persistence.GetDomainRequest{Name: request.Info.Name})
-	if err == nil {
-		// The domain already exists.
-		return nil, &workflow.DomainAlreadyExistsError{
-			Message: fmt.Sprintf("Domain already exists.  DomainId: %v", resp.Info.ID),
+func (m *sqlMetadataManagerV2) CreateDomain(request *persistence.CreateDomainRequest) (*persistence.CreateDomainResponse, error) {
+	// Encode request.Info.Data
+	data, err := gobSerialize(request.Info.Data)
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode DomainInfo.Data. Error: %v", err),
 		}
 	}
 
-	switch err.(type) {
-	case *workflow.EntityNotExistsError:
-		// Domain does not already exist. Create it.
-
-		// Encode request.Info.Data
-		data, err := gobSerialize(request.Info.Data)
-		if err != nil {
-			return nil, &workflow.InternalServiceError{
-				Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode DomainInfo.Data. Error: %v", err),
-			}
-		}
-
-		// Encode request.ReplicationConfig.Clusters
-		clusters, err := gobSerialize(persistence.SerializeClusterConfigs(request.ReplicationConfig.Clusters))
-		if err != nil {
-			return nil, &workflow.InternalServiceError{
-				Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode ReplicationConfig.Clusters. Error: %v", err),
-			}
+	// Encode request.ReplicationConfig.Clusters
+	clusters, err := gobSerialize(persistence.SerializeClusterConfigs(request.ReplicationConfig.Clusters))
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode ReplicationConfig.Clusters. Error: %v", err),
 		}
+	}
 
-		metadata, err := m.GetMetadata()
-		if err != nil {
-			return nil, err
-		}
+	metadata, err := m.GetMetadata()
+	if err != nil {
+		return nil, err
+	}
 
-		tx, err := m.db.Beginx()
-		if err != nil {
-			return nil, err
-		}
-		defer tx.Rollback()
+	tx, err := m.db.Beginx()
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
```

Comment:
[optional] I believe rollback is un-necessary when err is nil i.e. we successfully committed this transaction. Don't think it affects correctness - its might result in a additional call to the dbserver that results in a noop